src/lib.rs

```diff
@@ -127,12 +127,30 @@ impl RootContext for RootHandler {
             }
             .unwrap_or_default();
 
-            // host_addr format: service.namespace.svc.cluster.local
-            // cluster.local portion can be different based on which
-            // cluster the service is in
-            // But for now we're interesting about service and namespace only
-            let tokens: Vec<&str> = host_addr.split(".").collect();
-            let key = format!(KEY_FORMAT!(), tokens[1], tokens[0]);
+            let mut key = String::new();
+
+            if host_addr.is_empty()
```

Comment:
Nit but let's keep the if/else syntax consistent with the rest of the file

if {

} else {

}

Reply:
good call